modules/consensus/subscribe.go

```diff
@@ -169,12 +172,17 @@ func (cs *ConsensusSet) managedInitializeSubscribe(subscriber modules.ConsensusS
 		cs.mu.RLock()
 		err = cs.db.View(func(tx *bolt.Tx) error {
 			for i := 0; i < 100 && exists; i++ {
-				cc, err := cs.computeConsensusChange(tx, entry)
-				if err != nil {
-					return err
+				select {
+				case <-cancel:
+					return errors.New("aborting managedInitializeSubscribe")
```

Comment:
Yeah I wasn't sure about that one. I thought that since most of the time the channel that is passed is the `StopChan` of a threadgroup it might make sense to abort instantly right after calling `ConsensusSetSubscribe` instead of executing some other code like subscribing to the tpool which might take a few more seconds. Of course semantically it would make more sense to return `nil`.